src/hazelcore/layout/DHDoc.re

```diff
@@ -439,6 +482,24 @@ module Exp = {
           failwith("unexpected FailedCast without inner cast")
         | InvalidOperation(operation) =>
           switch (operation) {
+          | FailedSubscript(Subscript(d1, d2, d3)) =>
```

Comment:
`InvalidOperation` should take an operation and an error message or error class, so it will replace `FailedSubscript`, not wrap around it.

Reply:
Yunsoo and I have updated `InvalidOperation`. I generalize the runtime error decoration as `InvalidOpDecoration`. Other bugs are fixed.